net/instaweb/rewriter/css_filter_test.cc

```diff
@@ -2427,6 +2427,46 @@ TEST_F(CssFilterTest, AbsolutifyServingFallback) {
   TestFallbackFetch(url, expected_output);
 }
 
+TEST_F(CssFilterTest, BasicCsp) {
+  EnableDebug();
+  SetResponseWithDefaultHeaders("styles/a.css",
+                                kContentTypeCss, kInputStyle, 100);
+  SetResponseWithDefaultHeaders("uploads/sneaky.png",
+                                kContentTypeCss, kInputStyle, 100);
+
+  const char kCsp[] = "<meta http-equiv=\"Content-Security-Policy\" "
+                      "content=\"style-src */styles/ \">";
+  ValidateExpected(
+      "basic_csp",
+      StrCat(kCsp,
+             CssLinkHref("styles/a.css"),
+             CssLinkHref("uploads/sneaky.png")),
+      StrCat(kCsp,
+             CssLinkHref(Encode("styles/", "cf", "0", "a.css", "css")),
+             CssLinkHref("uploads/sneaky.png"),
+             "<!--The preceding resource was not rewritten "
+             "because CSP disallows its fetch-->"));
+}
+
+TEST_F(CssFilterTest, InlineCsp) {
+  EnableDebug();
+  options()->ClearSignatureForTesting();
+  options()->EnableFilter(RewriteOptions::kRewriteStyleAttributes);
+
+  const char kCsp[] = "<meta http-equiv=\"Content-Security-Policy\" "
+                      "content=\"style-src */styles/ \">";
+  const char kCss[] = "<style>* { display: stylish; }</style>";
+  const char kStyledDiv[] = "<div style='background-color: #f00; '/>";
+
+  ValidateExpected(
+      "inline_css",
+      StrCat(kCsp, kCss, kStyledDiv),
+      StrCat(kCsp, kCss,
+             "<!--Avoiding modifying inline style with CSP present-->",
```

Comment:
WDYT about moving the string constant in the impl file into a class static and referencing it here?